### PR TITLE
Add PHP 8 compatibility

### PR DESCRIPTION
**Pull request by alexander-schranz**
- [x] Add Test Matrix (Github Actions)

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              uses: ramsey/composer-install@v1
+              with:
+                  dependency-versions: ${{ matrix.dependency-versions }}
+                  composer-options: ${{ matrix.composer-options }}
+
+            - name: Bootstrap test environment
+              run: composer bootstrap-test-environment
+
+            - name: Execute test
+              if: matrix.coverage == false
```
Comment (alexander-schranz) [inline .github/workflows/test-application.yaml:80]: any hint to check for false here? tried different things with and without ${{ }} but nothing did work /cc @luca-rath

Comment (alexander-schranz) [reply .github/workflows/test-application.yaml:80]: okay did now work this way 🙈

File: .coveralls.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+service_name: github-actions
```
Comment (alexander-schranz) [inline .coveralls.yml]: Copied from: https://github.com/stripe/stripe-php/blob/master/.coveralls.github-actions.yml

File: .github/workflows/test-application.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+            - name: Execute test with coverage
+              if: ${{ matrix.coverage }}
+              run: composer test-with-coverage
+
+            - name: Lint Code
+              if: ${{ matrix.lint }}
+              run: composer lint
+
+            - name: Coverage all upload
```
Comment (alexander-schranz) [inline .github/workflows/test-application.yaml]: See https://github.com/php-coveralls/php-coveralls#github-actions

File: .github/workflows/test-application.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+            - name: Execute test
+              if: matrix.coverage == false
+              run: composer test
+
+            - name: Execute test with coverage
+              if: ${{ matrix.coverage }}
+              run: composer test-with-coverage
+
+            - name: Lint Code
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml]: think we executed the lint script before the test script in other repositories

File: .github/workflows/test-application.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            - name: Install composer dependencies
+              uses: ramsey/composer-install@v1
+              with:
+                  dependency-versions: ${{ matrix.dependency-versions }}
+                  composer-options: ${{ matrix.composer-options }}
+
+            - name: Bootstrap test environment
+              run: composer bootstrap-test-environment
+
+            - name: Execute test
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml]:
```suggestion
            - name: Execute test cases
```

File: .github/workflows/test-application.yaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+              if: matrix.coverage == false
+              run: composer test
+
+            - name: Execute test with coverage
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml]:
```suggestion
            - name: Execute test cases with coverage
```

File: .github/workflows/test-application.yaml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+            - name: Coverage all upload
+              if: ${{ matrix.coverage }}
+              run: |
+                  composer global require php-coveralls/php-coveralls
+                  php-coveralls -vvv
+              env:
+                  COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Check coverage
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml]:
```suggestion
            - name: Check code coverage
```

File: .github/workflows/test-application.yaml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+            - name: Execute test with coverage
+              if: ${{ matrix.coverage }}
+              run: composer test-with-coverage
+
+            - name: Lint Code
+              if: ${{ matrix.lint }}
+              run: composer lint
+
+            - name: Coverage all upload
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml]:
```suggestion
            - name: Upload code coverage to coveralls
```

File: .github/workflows/test-application.yaml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+              if: ${{ matrix.coverage }}
+              run: |
```
Comment (niklasnatter) [inline .github/workflows/test-application.yaml:92]: think i would use the official Coveralls github action for uploading 😇 https://github.com/marketplace/actions/coveralls-github-action

File: .github/workflows/test-application.yaml (hunk 4 above)
Comment (alexander-schranz) [reply .github/workflows/test-application.yaml]: did move it after the tests so the tests are always executed and linting of code style is done after it 🤔

File: .github/workflows/test-application.yaml (hunk 9 above)
Comment (alexander-schranz) [reply .github/workflows/test-application.yaml:92]: @nnatter that doesn't support PHPUNIT xml reports

Comment (alexander-schranz) [reply .github/workflows/test-application.yaml:92]: it supports just the `lcov` / `./coverage/lcov.info` format. So basically not PHP support which can only xml, json, html out of the box..

Open Issue: https://github.com/coverallsapp/github-action/issues/15

File: .github/workflows/test-application.yaml (hunk 4 above)
Comment (alexander-schranz) [reply .github/workflows/test-application.yaml]: anyway moved it up ;)

File: .coveralls.yml [hunk 10]
```diff
@@ -0,0 +1,3 @@
...
+coverage_clover: Tests/reports/clover.xml
```
Comment (niklasnatter) [inline .coveralls.yml]: think we could add these as argument to the `php-coveralls` call. but is okay for me this way too
https://github.com/coverallsapp/github-action/issues/15#issuecomment-643747251

File: .github/workflows/test-application.yaml (hunk 9 above)
Comment (niklasnatter) [reply .github/workflows/test-application.yaml:92]: you are right - sorry did not see that

**Review by niklasnatter — APPROVED** [review]

File: .coveralls.yml (hunk 10 above)
Comment (alexander-schranz) [reply .coveralls.yml]: I will try it if it works not sure the `service_name` can not be set and maybe thats required in this case that the PR is correctly updated, lets have a look if it works or not without the config.

**Pull request merged**